File: RangeSumQuery/rsq.java
class NumArray {
    private int[] data ;
    public NumArray(int[] nums) {
        if(nums.length == 0) return ;
        data = new int[nums.length];
        data[0] = nums[0];
        for(int i = 1; i<nums.length;i++){
            data[i] = nums[i] + data[i-1];
        }
    }
    
    public int sumRange(int i, int j) {
        return i==0?data[j]:data[j]-data[i-1];
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(i,j);
 */

File: RangeSumQuery/rsq.py
class NumArray:

    def __init__(self, nums: List[int]):
        if len(nums)==0:return 
        self.data = []
        self.data.append(nums[0])
        for i in range(1,len(nums)):
            self.data.append(nums[i] + self.data[i-1])

    def sumRange(self, i: int, j: int) -> int:
        return self.data[j] - self.data[i-1] if i!= 0 else self.data[j]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(i,j)
